docs/resources/iotda_dataforwarding.md

```diff
@@ -0,0 +1,224 @@
+is `DMS_KAFKA_FORWARDING`. The [properties](#IoTDA_kafka_forwarding) structure is documented below.
+
+<a name="IoTDA_http_forwarding"></a>
+The `http_forwarding` block supports:
+
+* `url` - (Required, String) Specifies the Push URL. The request method must is post.
+
+<a name="IoTDA_dis_forwarding"></a>
+The `dis_forwarding` block supports:
+
+* `region` - (Required, String) Specifies the region which the DIS stream belongs to.
+
+* `project_id` - (Required, String) Specifies the project ID which the DIS stream belongs to.
+
+* `stream_id` - (Required, String) Specifies the DIS stream ID.
+
+<a name="IoTDA_obs_forwarding"></a>
+The `obs_forwarding` block supports:
+
+* `region` - (Required, String) Specifies the region which the OBS belongs to.
+
+* `project_id` - (Required, String) Specifies the project ID which the OBS belongs to.
+
+* `bucket` - (Required, String) Specifies the OBS Bucket.
+
+* `custom_directory` - (Optional, String) Specifies the custom directory for storing channel files. The ID contains a
+ maximum of 256 characters. Multi-level directories can be separated by (/), and cannot start or end with a slash (/),
+ and cannot contain more than two adjacent slashes (/). Only letters, digits, hyphens (-), underscores (_), slash (/)
+ and braces ({}) are allowed. Braces can be used only for the time template parameters. For example, if the custom
+ directory is in the format of {YYYY}/{MM}/{DD}/{HH}, data is generated in the directory based on the current
+ time(for example, 2022/06/14/10) when data is forwarded.
+
+<a name="IoTDA_amqp_forwarding"></a>
+The `amqp_forwarding` block supports:
+
+* `queue_name` - (Required, String) Specifies the AMQP Queue name.
+
+<a name="IoTDA_kafka_forwarding"></a>
+The `kafka_forwarding` block supports:
+
+* `region` - (Required, String) Specifies the region which the KAFKA belongs to.
+
+* `project_id` - (Required, String) Specifies the project ID which the KAFKA belongs to.
+
+* `topic` - (Required, String) Specifies the topic.
+
+* `addresses` - (Required, List) Specifies the list of the connected service addresses.
+The [addresses](#IoTDA_forwarding_addresses) structure is documented below.
+
+* `user_name` - (Optional, String) Specifies the SASL user name.
+
+* `password` - (Optional, String) Specifies the password.
+
+<a name="IoTDA_forwarding_addresses"></a>
+The `addresses` block supports:
+
+* `ip` - (Optional, String) Specifies the IP of the connected service address.
+Exactly one of `ip` or `domain` must be provided.
+
+* `domain` - (Optional, String) Specifies the domain of the connected service address.
+Exactly one of `ip` or `domain` must be provided.
+
+* `port` - (Required, Int) Specifies the port of the connected service address.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID in UUID format.
+
+* `targets` - all the targets attributes:
```

Comment:
`targets` is already included in arguments above

Reply:
ok